tests/js/client/aql/aql-mdi.js

```diff
@@ -309,19 +383,139 @@ function optimizerRuleMdi2dIndexTestSuite() {
     testCreateAsZkd: function () {
       let col = db._create(colName + "5");
       const idx = col.ensureIndex({
-        type: 'zkd',
+        type: "zkd",
         name: "myIndex",
-        fields: ['x', 'y'],
-        fieldValueTypes: 'double',
+        fields: ["x", "y"],
+        fieldValueTypes: "double",
       });
-      assertEqual('zkd', idx.type);
+      assertEqual("zkd", idx.type);
 
       const idx2 = col.index("myIndex");
       assertEqual(idx2.id, idx.id);
-      assertEqual(idx2.type, 'zkd');
+      assertEqual(idx2.type, "zkd");
 
       col.drop();
-    }
+    },
+
+    testInRangeNonStrict: function () {
+      const queries = [
+        aql`FOR d IN ${col} FILTER 0 < d.x && d.x < 1 RETURN d.x`,
+        aql`FOR d IN ${col} FILTER IN_RANGE(d.x, 0, 1, false, false) RETURN d.x`,
+      ];
+
+      for (let i = 0; i < queries.length; ++i) {
+        const query = queries[i];
+        const explainRes = db
+          ._createStatement({ query: query.query, bindVars: query.bindVars })
+          .explain();
+        const appliedRules = explainRes.plan.rules;
+        const nodeTypes = explainRes.plan.nodes
+          .map((n) => n.type)
+          .filter((n) => !["GatherNode", "RemoteNode"].includes(n));
+        assertEqual(["SingletonNode", "IndexNode", "ReturnNode"], nodeTypes);
+        assertTrue(appliedRules.includes(useIndexes));
+        assertFalse(appliedRules.includes(removeFilterCoveredByIndex));
+        const executeRes = db._query(query.query, query.bindVars);
+        const res = executeRes.toArray();
+        res.sort();
+        assertEqual([0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9], res);
+      }
+    },
+
+    testInRangeLeftStrict: function () {
+      const queries = [
+        {
+          q: aql`FOR d IN ${col} FILTER 0 <= d.x && d.x < 1 RETURN d.x`,
+          inRange: false,
+        },
+        {
+          q: aql`FOR d IN ${col} FILTER IN_RANGE(d.x, 0, 1, true, false) RETURN d.x`,
+          inRange: true,
+        },
+      ];
+
+      for (let i = 0; i < queries.length; ++i) {
+        const query = queries[i].q;
+        const explainRes = db
+          ._createStatement({ query: query.query, bindVars: query.bindVars })
+          .explain();
+        const appliedRules = explainRes.plan.rules;
+        const nodeTypes = explainRes.plan.nodes
+          .map((n) => n.type)
+          .filter((n) => !["GatherNode", "RemoteNode"].includes(n));
+        assertEqual(["SingletonNode", "IndexNode", "ReturnNode"], nodeTypes);
+        if (queries[i].inRange) {
+          assertFalse(appliedRules.includes(removeFilterCoveredByIndex));
+        } else {
+          assertTrue(appliedRules.includes(removeFilterCoveredByIndex));
+        }
```

Comment:
same but the other way around. we want to ensure that there is a post filter on the index node, and that it is precisely the condition we need, i.e. `IN_RANGE(...)`.